File: pgm/ABC_C#/ABC_135_A.cs
using System;
// using System.Collections.Generic;
// using System.Globalization;
// using System.IO;
// using System.Linq;
// using System.Text;
// using System.Text.RegularExpressions

class Program
{
    static void Main()
    {
        string[] arr = Console.ReadLine().Split();
        int A, B;
        A = int.Parse(arr[0]);
        B = int.Parse(arr[1]);
        Console.WriteLine((A + B) % 2 == 0 ? ((A + B) / 2).ToString() : "IMPOSSIBLE");
        return;
    }
}

File: pgm/ABC_Visual_Basic/ABC_135_A.vb
' Imports System.Collections.Generic
' Imports System.Globalization
' Imports System.IO
' Imports System.Linq
' Imports System.Text
' Imports System.Text.RegularExpressions

Class Main
    Public Shared Sub Main()
        Dim A, B As Integer
        Dim arr() As String
        arr = Split(Console.ReadLine())
        A = arr(0) : B = arr(1)
        Console.WriteLine(If((A + B) Mod 2 = 0, (A + B) / 2, "IMPOSSIBLE"))
        Return
    End Sub
End Class